### DISCUSSION TITLE
Soft delete notifications

### DISCUSSION THREAD

**Discussion by studiowizjo** [filamentphp/filament#4147] [Help] [answered]
Is there any clever way to soft delete notifications (for now they are permanently deleted)? I added column in database, but I cannot use `SoftDeletes` trait in model as it is inner Laravel model. Of course I can extend class, but don`t know how to "inform" Filament to use my class instead.

I thought about binding class in `AppServiceProvider`, but without success:

```php
$this->app->bind(Notification::class, DatabaseNotification::class);
```

The other possibility could be hooking somehow into `clearDatabaseNotifications()` Livewire method.

The last option, which I I'd rather avoid is to publish view and then switch above method to mine.

Any other ideas?

**Comment by sh4msi** [comment]
@studiowizjo Did you find a standard solution for this?

**Reply by studiowizjo** [reply]
Unfortunately not. Besides above i tried also decorating class, similiar to the[ example in docs](https://laravel.com/docs/9.x/container#extending-bindings):

```php
$this->app->extend(Service::class, function ($service, $app) {
    return new DecoratedService($service);
});
```
but without success. For now I don`t have any other ideas =/.

**Comment by sh4msi** [comment, marked as answer]
This is how I solved it, I hope it will be useful for you and other friends

.
_App\Classes\MyDatabaseNotification.php_
```
namespace App\Classes;

use Illuminate\Database\Eloquent\SoftDeletes;
use Illuminate\Notifications\DatabaseNotification;

class MyDatabaseNotification extends DatabaseNotification
{
    use SoftDeletes;

    protected $dates = ['deleted_at'];
}
```


_App\Classes\MyNotifiable.php_
```
namespace App\Classes;

use Illuminate\Notifications\Notifiable;

trait MyNotifiable
{
    use Notifiable;

    /**
     * Get the entity's notifications.
     */
    public function notifications()
    {
        return $this->morphMany(MyDatabaseNotification::class, 'notifiable')
            ->orderBy('created_at', 'desc');
    }
}
```


_App\User.php_
```
namespace App\Models;

use Illuminate\Foundation\Auth\User as Authenticatable;
use App\Classes\MyNotifiable;
...

class User extends Authenticatable
{
    use MyNotifiable;
  ...
```

**Reply by studiowizjo** [reply]
Works :). Nice idea!

**Reply by KhairulAzmi21** [reply]
its this the optimize solution ?
